contracts/PolyptychRandomizerV0.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Creatd By: Art Blocks Inc.
+
+pragma solidity 0.8.17;
+
+import "./interfaces/0.8.x/IRandomizerV2.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "./interfaces/0.8.x/IFilteredMinterPolyptychV0.sol";
+
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+
+contract PolyptychRandomizerV0 is IRandomizerV2, Ownable {
+    // The core contract that may interact with this randomizer contract.
+    IGenArt721CoreContractV3 public genArt721Core;
+
+    // Used to obtain the project ID from the token ID
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    // The Polyptych randomizer requires a minter that provides a seed token ID for new tokens
+    address public polyptychMinterAddress;
+
+    // The second/subsequent panels of polyptychs should be configured to bypass randomizing
+    mapping(uint256 => bool) _projectIsPolyptych;
+
+    function assignCoreAndRenounce(address _genArt721Core) external onlyOwner {
+        renounceOwnership();
+        genArt721Core = IGenArt721CoreContractV3(_genArt721Core);
+    }
+
+    // Allows the owner of the core contract to change the polyptych minter
+    function setPolyptychMinterAddress(address _polyptychMinterAddress)
+        external
+    {
+        require(msg.sender == genArt721Core.owner(), "Only admin may call");
+        polyptychMinterAddress = _polyptychMinterAddress;
+    }
+
+    // Allows the owner of the core contract to configure a project as a polyptych
+    function toggleProjectIsPolyptych(uint256 _projectId) external {
+        require(msg.sender == genArt721Core.owner(), "Only admin may call");
+        _projectIsPolyptych[_projectId] = !_projectIsPolyptych[_projectId];
+    }
+
+    // Allows the current configuration of a project as a polyptych to be easily queried
+    function projectIsPolyptych(uint256 _projectId)
+        external
+        view
+        returns (bool)
+    {
+        return _projectIsPolyptych[_projectId];
+    }
+
+    // When `genArt721Core` calls this, it can be assured that the randomizer
+    // will set a bytes32 hash for tokenId `_tokenId` on the core contract.
+    function assignTokenHash(uint256 _tokenId) external virtual {
+        require(msg.sender == address(genArt721Core), "Only core may call");
+
+        if (_projectIsPolyptych[_tokenId / ONE_MILLION]) {
+            bytes32 seededHash = IFilteredMinterPolyptychV0(
+                polyptychMinterAddress
+            ).polyptychSeedForToken(_tokenId);
```

Comment:
Ensure this function call fails if the method does not exist on the minter contract.